### PR TITLE
Use null as the ZIP 317 fee sentinel instead of -1

### LINKED ISSUE #6556: Use JSON null instead of -1 as the sentinel for a ZIP 317 conventional fee
This is less error-prone, since a null can't be accidentally treated as a number or have arithmetic done on it.

Since the code using the -1 sentinel was ~~only recently merged~~ in #6524, we should do this before it appears in the 5.5.0 release, to avoid needing to accept -1 as well for compatibility.

### PR DESCRIPTION
**Pull request by sellout**
Fixes #6556.

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -5012,7 +5012,7 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
     int nMinDepth = parseMinconf(DEFAULT_NOTE_CONFIRMATIONS, params, 2, std::nullopt);
 
     std::optional<CAmount> nFee;
-    if (params.size() > 3 && params[3].get_real() != -1.0) {
+    if (params.size() > 3 && !params[3].isNull()) {
```
Comment (str4d) [inline src/wallet/rpcwallet.cpp:5015]: How do we specify `null` when calling this via `zcash-cli`? Is it the unquoted string `null`?

Comment (daira) [reply src/wallet/rpcwallet.cpp:5015]: I believe so.

**Review by daira — APPROVED** [review]
utACK. Please add an example using null and a privacy policy, and another using an explicit fee, to the help.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (nuttycom) [reply src/wallet/rpcwallet.cpp:5015]: I'll try this out locally.

**Review by nuttycom — APPROVED** [review]
utACK

**Review by nuttycom — CHANGES_REQUESTED** [review]
I tried to test this locally, but the following failed: 
```
./src/zcash-cli z_sendmany "utest1a60h665wp0psjnmf34q7w2e8ej48djt667phars97x6ewsnxalwz9a4z9kn7u46k95p6wq8rrqydx2eh64lxdaa3w435wnz0mc59ymx4" \
  '[{"address": "utest12yxahhtv27jhhrwt458fjgu8lwczaqtzquyvmvg7wdxwn7gl5lq3qruce9r7g6rzry7fr60ew3yqnu3zcsr0w2sanrq9gay2hfa7t0gttgdhrr79m48e9lae8w2tacuhs4jtgu4ee9k", "amount": 0.001}]' \
  null "AllowRevealedSenders"
```
Failed with
```
nuttycom@sword:~/work/zcash/release (conventional-fee-sentinel)λ . ../canon/nuttycom/test_many.sh
zcash-cli: /home/nuttycom/work/zcash/release/depends/x86_64-pc-linux-gnu/share/../include/tl/expected.hpp:1937: E &tl::expected<UniValue, UniValue>::error() & [T = UniValue, E = UniValue]: Assertion `!has_value()' failed.
Aborted (core dumped)
```
Both quoted and unquoted `null` fail here in the same fashion. Leaving off the last two parameters, the call succeeds.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (sellout) [reply src/wallet/rpcwallet.cpp:5017]: Yep – it’s already working and testable in other operations (like `setban`).

**Comment by sellout** [comment]
> I tried to test this locally, but the following failed:

After fixing #6561, I get the following error from your command “Error parsing JSON: AllowRevealedSenders” – this is an off-by-one in the provided arguments. Adding a `minConf` value before the `null` works (although the server does tell me “no payment source found” 😁).

**Review by nuttycom — APPROVED** [review]
Tested ACK

**Pull request merged**